ase/rtl/CMakeLists.txt

```diff
@@ -0,0 +1,153 @@
+## Copyright(c) 2017, 2018, Intel Corporation
+##
+## Redistribution  and  use  in source  and  binary  forms,  with  or  without
+## modification, are permitted provided that the following conditions are met:
+##
+## * Redistributions of  source code  must retain the  above copyright notice,
+##   this list of conditions and the following disclaimer.
+## * Redistributions in binary form must reproduce the above copyright notice,
+##   this list of conditions and the following disclaimer in the documentation
+##   and/or other materials provided with the distribution.
+## * Neither the name  of Intel Corporation  nor the names of its contributors
+##   may be used to  endorse or promote  products derived  from this  software
+##   without specific prior written permission.
+##
+## THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+## AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING,  BUT NOT LIMITED TO,  THE
+## IMPLIED WARRANTIES OF  MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+## ARE DISCLAIMED.  IN NO EVENT  SHALL THE COPYRIGHT OWNER  OR CONTRIBUTORS BE
+## LIABLE  FOR  ANY  DIRECT,  INDIRECT,  INCIDENTAL,  SPECIAL,  EXEMPLARY,  OR
+## CONSEQUENTIAL  DAMAGES  (INCLUDING,  BUT  NOT LIMITED  TO,  PROCUREMENT  OF
+## SUBSTITUTE GOODS OR SERVICES;  LOSS OF USE,  DATA, OR PROFITS;  OR BUSINESS
+## INTERRUPTION)  HOWEVER CAUSED  AND ON ANY THEORY  OF LIABILITY,  WHETHER IN
+## CONTRACT,  STRICT LIABILITY,  OR TORT  (INCLUDING NEGLIGENCE  OR OTHERWISE)
+## ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,  EVEN IF ADVISED OF THE
+## POSSIBILITY OF SUCH DAMAGE.
+
+cmake_minimum_required(VERSION 2.8.12)
+project("opae-c-ase-platform")
+
+# find afu_sim_setup
+find_program(AFU_PLATFORM_CONFIG
```

Comment:
What's going on with this find_program? afu_platform_config should be in OPAE SDK bin.

Reply:
right; but user can find afu_platform_config under /usr/local/bin or somewhere else. so truly you should need a find_program. this is standard practice in cmake, you dont call executables directly with their names, not even gcc/make/cmake/vsim/vlog... etc.

Reply:
Ah, I see. You might be better removing the additional PATHS and just use the default PATH in order to match the Quartus flow.

A better question is, why are you invoking afu_platform_config at all here? I suspect that most of the other problems in this CMakeLists come from trying to invoke afu_platform_config here instead of letting generate_ase_environment compile the set of sources for you that you will then import into CMake. There are lots of dynamic decisions made in the Python that became static in this example.